### ISSUE TITLE
Related links must be allowed inside relationship documents

### ISSUE THREAD

**Issue by alexpkalinka** [json-api-php/json-api#104]
In JSON:API docs the "[Fetching Relationships](https://jsonapi.org/format/#fetching-relationships-responses-200)" section says:
> The top-level links object MAY contain self and related links, as described above for relationship objects.

And it gives the example response:
```json
{
  "links": {
    "self": "/articles/1/relationships/author",
    "related": "/articles/1/author"
  },
  "data": {
    "type": "people",
    "id": "12"
  }
}
```

However, it seems like this library does not allow to generate such document. Here's my attempt:
```php
new JsonApi\DataDocument(
    new JsonApi\ResourceIdentifier("people", "12"),
    new JsonApi\Link\SelfLink("/articles/1/relationships/author")
);
```

It gives:
```json
{
    "data": {
        "type": "people",
        "id": "12"
    },
    "links": {
        "self": "/articles/1/relationships/author"
    }
}
```

But the `related` link can't be added because `DataDocument` constructor does not accept `RelatedLink` objects.

**Comment by f3ath** [comment]
Thank you. The bugfix has been released as v2.1.2. Please confirm if it works for you.

Usage examples: 
- https://github.com/json-api-php/json-api/blob/master/test/DataDocument/SingleResourceIdentifierTest.php
- https://github.com/json-api-php/json-api/blob/master/test/DataDocument/ManyResourceIdentifiersTest.php

**Comment by alexpkalinka** [comment]
Wow, this was fast!
Yes, it works correctly now. Thank you.
